pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -498,43 +501,57 @@ func maybeRevertToCutoverTimestamp(
 	p := execCtx.(sql.JobExecContext)
 	db := p.ExecCfg().DB
 	jobRegistry := p.ExecCfg().JobRegistry
-	j, err := jobRegistry.LoadJob(ctx, ingestionJobID)
+	ingestionJob, err := jobRegistry.LoadJob(ctx, ingestionJobID)
 	if err != nil {
 		return false, err
 	}
-	details := j.Details()
-	var sd jobspb.StreamIngestionDetails
-	var ok bool
-	if sd, ok = details.(jobspb.StreamIngestionDetails); !ok {
-		return false, errors.Newf("unknown details type %T in stream ingestion job %d",
-			details, ingestionJobID)
-	}
-	progress := j.Progress()
-	var sp *jobspb.Progress_StreamIngest
-	if sp, ok = progress.GetDetails().(*jobspb.Progress_StreamIngest); !ok {
-		return false, errors.Newf("unknown progress type %T in stream ingestion job %d",
-			j.Progress().Progress, ingestionJobID)
-	}
 
-	cutoverTime := sp.StreamIngest.CutoverTime
-	if cutoverTime.IsEmpty() {
-		log.Infof(ctx, "empty cutover time, no revert required")
-		return false, nil
+	var isRevertRequired bool
```

Comment:
The name could be a bit better here as this is less about whether a revert is _required_ and more about whether it is possible yet.

Reply:
renamed to `shouldRevertToCutover`, but I'm open for suggestions..